src/captions.rs

```diff
@@ -193,50 +194,70 @@ pub async fn generate_captions(
         });
     }
 
-    let source = match Url::parse(&options.input) {
-        Ok(_) => Ok(AudioSource::from_url(&options.input)),
-        Err(url_error) => {
-            debug!("url failed to parse {:?}", url_error);
-            let filepath =
-                Utf8PathBuf::from(&options.input);
-            let file = File::open(&filepath).await.unwrap();
+    let filepath = Utf8PathBuf::from(&options.input);
 
-            match mime_guess::from_path(&options.input)
-                .first()
-            {
-                Some(guess) => {
-                    if guess.type_() != "audio" {
-                        Err(CaptionError::InvalidMimeType {
+    let response = if filepath.clone().extension().unwrap()
+        == "json"
+    {
+        let result = fs::read_to_string(filepath).unwrap();
+        serde_json::from_str(&result).unwrap()
+    } else {
+        let source = match Url::parse(&options.input) {
+            Ok(_) => {
+                Ok(AudioSource::from_url(&options.input))
+            }
+            Err(url_error) => {
+                debug!(
+                    "url failed to parse {:?}",
+                    url_error
+                );
+
+                let file =
+                    File::open(&filepath).await.unwrap();
+                match mime_guess::from_path(&options.input)
+                    .first()
+                {
+                    Some(guess) => {
+                        if guess.type_() != "audio" {
+                            Err(CaptionError::InvalidMimeType {
                             guess,
                         })
-                    } else {
-                        Ok(AudioSource::from_buffer_with_mime_type(
+                        } else {
+                            Ok(AudioSource::from_buffer_with_mime_type(
                             file,
                             guess.to_string(),
                         ))
+                        }
+                    }
+                    None => {
+                        Err(CaptionError::MimeGuessError {
+                            filepath: filepath,
+                        })
                     }
                 }
-                None => Err(CaptionError::MimeGuessError {
-                    filepath: filepath,
-                }),
             }
-        }
-    }?;
+        }?;
 
-    let dg_client =
-        Deepgram::new(&options.deepgram_api_key);
+        let dg_client = Deepgram::new(
+            options
+                .deepgram_api_key
+                .as_ref()
+                .expect("deepgram API Key is required"),
```

Comment:
I imagine we can handle this error more cleanly, but I was having trouble wiring up the Error codes. Open to recommendations.

Reply:
fine for now. It would kill the program anyway even if we returned the error. The error code would have to be added to the deepgram error before being returnable. Likely just a manual construction and return of the error variant.